pkg/webservice/handlers.go

```diff
@@ -521,3 +521,39 @@ func updateConfiguration(conf string) (string, error) {
 	}
 	return "", fmt.Errorf("config plugin not found")
 }
+
+func getPartitions(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+
+	var partitionsInfo []*dao.PartitionInfo
+	lists := schedulerContext.GetPartitionMapClone()
+	for _, partitionContext := range lists {
+		partitionInfo := &dao.PartitionInfo{}
+		partitionInfo.Name = partitionContext.Name
+		partitionInfo.State = partitionContext.StateMachine.Current()
```

Comment:
It is not a good idea to expose `StateMachine` and `StateTime`, they are pretty internal objects.
Instead, we can create a `GetCurrentState()` function that returns a string. Same for the stateTime.

Reply:
Taken care.